nginx.master.conf

```diff
@@ -156,21 +156,29 @@ http {
             proxy_pass http://navstar/lashup/key;
         }
 
+        location ~ ^/(slave|agent)/(?<agentid>[0-9a-zA-Z-]+)$ {
+            # Append slash and perform internal-redirect.
+            rewrite ^/(slave|agent)/(.*)$ /agent/$2/ last;
+        }
+
         location ~ ^/(slave|agent)/(?<agentid>[0-9a-zA-Z-]+)(?<url>.*)$ {
```

Comment:
Totally. I have made this change as we agreed in dcos-sec-team channel, but it must have been lost during rebasing :(